File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float n; // number of change owed
    int left; // how much charge is left, in cents, thus integer, we need integer (cents) to avoid floating-point number problems
    int coins = 0; // number of coins needed

    // Prompt user for the number of change owed
    // Prompt once and prompt again if the number is negative
    do
    {
        n = get_float("Change owed: ");
    }
    while (n < 0);

    left = round(n * 100);
    while (left > 0) // do until we owe any money
    {
        if (left - 25 >= 0) // try with the biggest coins aka quarters first
        {
            left -= 25;
        }
        else if (left - 10 >= 0) // if the biggest coins are too much try lowers, that is dimes
        {
            left -= 10;
        }
        else if (left - 5 >= 0) // if dimes are too much, try nickels
        {
            left -= 5;
        }
        else // use the lowest ones, that is pennies, if no one before was suitable
        {
            left -= 1;
        }
        coins++; // increase amount of used coins on every step
    }

    printf("%i\n", coins);
}

File: pset6/cash/cash.py
# Prompt user for the change owed
# Prompt again if it's a negative number
while True:
    n = input("Change owed: ")
    try:
        n = float(n)
        if n >= 0:
            break
    except ValueError:
        continue

# Change owed in cents
left = round(n * 100)

# Take integer division of left for 25
# Take integer division of the rest for 10
# The same for 5 and 1
# Add all
# That's the answer! :)
print(left // 25 + left % 25 // 10 + left % 25 % 10 // 5 + left % 25 % 10 % 5)